src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,25 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.supplyAsync(() -> cache.get(userId), executor);
+        }
+        UserData userData = new UserData(userId, "Details for user" + userId);
```

Comment:
We should also add some thread sleep when we put the data in the cache to simulate some DB access time. 
We can also have some `UserDetails getDetailsForUser(userId)` method where we can put all that logic and call it in the `computeIfAbsent()`